lib/chef/http/authenticator.rb

```diff
@@ -166,60 +178,85 @@ def self.get_cert_password
 
         present.each do |secret|
           if secret[:name] == "PfxPass"
-            return secret[:data]
+            password = decrypt_pfx_pass(secret[:data])
+            return password
           end
         end
 
-        # if we make it this far, that means there is no valid password in the Registry. Fail out to correct that.
         raise Chef::Exceptions::Win32RegKeyMissing
 
       rescue Chef::Exceptions::Win32RegKeyMissing
         # if we don't have a password, log that and generate one
-        Chef::Log.warn "Authentication Hive and value not present in registry, creating it now"
+        Chef::Log.warn "Authentication Hive and values not present in registry, creating them now"
         new_path = "HKEY_LOCAL_MACHINE\\Software\\Progress\\Authentication"
         unless @win32registry.key_exists?(new_path)
           @win32registry.create_key(new_path, true)
         end
         password = SOME_CHARS.sample(1 + rand(SOME_CHARS.count)).join[0...14]
-        values = { name: "PfxPass", type: :string, data: password }
+        encrypted_pass = encrypt_pfx_pass(password)
+        values = { name: "PfxPass", type: :string, data: encrypted_pass }
         @win32registry.set_value(new_path, values)
         password
       end
 
+      def self.encrypt_pfx_pass(password)
+        powershell_code = <<~CODE
```

Comment:
As discussed, I am punting for now since the ReservedNames code is interchangeable with the PowerShell approach. Taking a backlog item to do that work